src/kata6.joinNames.js

```diff
@@ -0,0 +1,11 @@
+const joinNames = (namesObj) => {
+
+ const newArray = namesObj.map(iterator => iterator.name);
+
+    newArray.splice(newArray.length-1, 1, '& '+ newArray[newArray.length-1])
```

Comment:
Assign `newArray.length - 1` to a variable with a meaningful name such as `lastElementIndex`. Also, use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals):

```js
const lastElementIndex = newArray.length - 1

newArray.splice(lastElementIndex, 1, `& ${newArray[lastElementIndex]}`)
```